From 12494337e4ce67ce927bd2d8c86e9c176ec3e36f Mon Sep 17 00:00:00 2001
From: Johan Hedberg <johan.hedberg@intel.com>
Date: Fri, 20 Sep 2013 11:17:34 -0500
Subject: [PATCH] core: Fix passing timeout when disabling discoverable mode

We should only pass a timeout value when enabling discoverable mode.
When disabling passing something non-zero would yield a "invalid
parameters" error.
---
 src/adapter.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/adapter.c b/src/adapter.c
index 17f5508..5434a07 100644
--- a/src/adapter.c
+++ b/src/adapter.c
@@ -1890,7 +1890,8 @@ static void property_set_mode(struct btd_adapter *adapter, uint32_t setting,
 	case MGMT_SETTING_DISCOVERABLE:
 		memset(&cp, 0, sizeof(cp));
 		cp.val = mode;
-		cp.timeout = htobs(adapter->discoverable_timeout);
+		if (cp.val)
+			cp.timeout = htobs(adapter->discoverable_timeout);
 
 		opcode = MGMT_OP_SET_DISCOVERABLE;
 		param = &cp;
-- 
1.8.3.1

